pkg/controller/service/service_controller.go

```diff
@@ -282,6 +282,11 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 	var err error
 
 	if !wantsLoadBalancer(service) {
+		if v1helper.LoadBalancerStatusEqual(previousState, &v1.LoadBalancerStatus{}) {
```

Comment:
Can we do `v1helper.LoadBalancerStatusEqual(previousState, &v1.LoadBalancerStatus{})` -> `v1helper.LoadBalancerStatusEqual(previousState, newState)` instead?

Reply:
Using a literal here seems more explicit and less prone to future changes in how `newState` is initialized or used, but maybe I'm overthinking it.

If I do change this to `v1helper.LoadBalancerStatusEqual(previousState, newState)`, should I also move the assignment `newState = &v1.LoadBalancerStatus{}` from the end of the outer `if` block to precede that `if`block? Strictly speaking, we don't even need that assignment, so I assume it's there to make the intention more explicit.

Reply:
Actually, that assignment is needed, and I would need to move it up in order to use it here.

Reply:
Good point :)